cmd/chat/index.ts

```diff
@@ -0,0 +1,251 @@
+        botMessageBackgroundColor: "rgba(220, 242, 247)",
+        userMessageColor: "#E2E8F0",
+        userMessageBackgroundColor: "#4A5568",
+        botName: "ChatBot",
+        buttonBackgroundColor: "#4A5568",
+        buttonBorderColor: "#2D3748",
+        buttonBorderWidth: "1px",
+        dotsColor: "#A0AEC0",
+        promptId: "feea76f3-af40-4df9-be47-ad073d504615",
+        memoryId: "",
+    },
+    footer: {
+        bgColor: "#2D3748",
+        textColor: "#E2E8F0",
+    },
+};
+
+function formatPrompt(options: Record<string, string>): string {
+    return `
+    From this short description:
+    "${options?.topic}"
+
+    Detect the topic from the above description and generate a theme with relevant and complementary colors and adding the colors in this object:
+    const config: Config = ${JSON.stringify(defaultConfig, null, 4)};
+
+    "chatBackgroundColor" must be the same color as "botMessageBackgroundColor" but with a few opacity.
+    However each color must be a complementary color but unique.
+
+    Answer only the object as if you were logging this command: \`console.log(JSON.parse(config))\`.
+
+    Assistant: Sure! Here is the theme in the config object:
+    `;
+}
+
+async function generateTheme(options: Record<string, string>, token: string): Promise<string> {
+    console.log("Generating a theme for the chatbot...");
+
+    const prompt = formatPrompt(options);
+    const { result } = await generateWithTokenUsage(
+        prompt,
+        { model: "gpt-4" },
+        { token, endpoint: "https://api2.polyfact.com" },
+    ).catch((e) => {
+        throw new Error(`Error during OpenAI request: ${e.message}`);
+    });
+
+    return result;
+}
+
+async function updateConfig(repo: string, res: string): Promise<void> {
+    const filePath = path.join(repo, "src/config.ts");
+    const data = await fs.readFile(filePath, "utf8");
+    const updatedContent = data.replace(
+        /\/\/ \[start\](.*?)\[end\]/gs,
+        `const config: Config = ${extractConfig(res)}`,
+    );
+
+    await fs.writeFile(filePath, updatedContent, "utf8");
+    console.log("Config file updated successfully with the new theme!");
+}
+
+const questions = [
+    {
+        type: "list",
+        name: "testBoilerplate",
+        message: "Would you like to test the boilerplate with predefined settings?",
+        choices: ["Yes", "No"],
+        default: "No",
+    },
+
+    {
+        type: "list",
+        name: "stack",
+        message:
+            "What stack would you like to use? (If you are not sure, please select the default option)",
+        choices: Object.keys(BOILERPLATES).map((key) => key),
+        default: BOILERPLATES[REACT_BOILERPLATE],
+        when: (answers: { testBoilerplate: string }) => answers.testBoilerplate === "No",
+    },
+    {
+        type: "list",
+        name: "style",
+        message:
+            "Would you like to use a preset style or generate with LLM a custom style based on a specific topic?",
```

Comment:
"Would you like to use a preset style or generate custom style based on a specific topic using LLMs ?"